gnome-image-installer/pages/install/gis-scribe.c

```diff
@@ -0,0 +1,820 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+/*
+ * Copyright (C) 2016-2017 Endless Mobile, Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
+ * 02111-1307, USA.
+ */
+#include "config.h"
+#include "gis-scribe.h"
+
+#include <errno.h>
+#include <glib/gi18n.h>
+#include <gio/gunixoutputstream.h>
+
+#include <sys/ioctl.h>
+/* for BLKGETSIZE64, BLKDISCARD */
+#include <linux/fs.h>
+
+#include "gduxzdecompressor.h"
+#include "glnx-errors.h"
+
+#define IMAGE_KEYRING "/usr/share/keyrings/eos-image-keyring.gpg"
+
+typedef struct _GisScribe {
+  GObject parent;
+
+  GFile *image;
+  guint64 image_size;
+  GFile *signature;
+  gchar *keyring_path;
+  gchar *drive_path;
+  gint drive_fd;
```

Comment:
Okay. Actually I don't think it's strictly necessary since once the worker thread is spawned it's only accessed on the main thread from `finalize`, by which point the thread is dead.

Reply:
Then it shouldn’t be accessed from within the mutux. Either is fine, but consistency is good.